contract.md

```diff
@@ -12,29 +17,60 @@ like to work on, as part of your project (for example, other things related to D
+The milestones are broken down week by week. Seeing that there are 5 weeks to work on this, I hope to spend my time accordingly:
+Week 1: Investigate benefits and tradeoffs between host languages, and determine whether an external/internal implementation will work best. Finalize the syntax/language design knowing this. Decide on a general architecture/structure of the DSL (I know that professor Wiedermann already discussed making two distinct DSL’s: One for letting the user create events using the DSL syntax, which can then could compile toiCalendar, and another for taking in that iCalendar input and showing a preview in either Excel or HTML/CSS. While this is very broad/vague, I hope by the end of week 1 to have this finalized). This should also require research on different calendar DSL’s such as iCalendar and understanding how an iCalendar file could be output using a language, whether through writing to a file manually or by using libraries in the host language.
+Week 2: Make progress in implementing the first DSL: Parsing user input. This will be my primary focus, implementing parsing combinators and reading input from a file/text stream. I hope by the end of this week I will have fully implemented the syntax for the user.
+Week 3: Finish the first DSL. I hope by the end of this week the DSL will be able to output to iCalendar or a specific filetype that could then be read into the second DSL. This will require some foresight to understand how the second DSL should read in input. Hopefully most of the design language and implementation of parsing user input will be done already. I am worried about exporting to iCalendar, as I do not know whether using a library (or if it exists!) or writing my own solution would be best. I hope to determine that in week 3 and to have a solution.
+Week 4: Work on the second DSL. Make progress in outputting to a HTML/CSS file. I believe that my work in the previous weeks should make this easier as much of the exploratory work should be done. Therefore, I hope that by the end of this week I will have the preview options and the second DSL nearly complete.
+Week 5: Finish the second DSL, finish the previews, link everything together. I am unsure how much time this will take, so I am leaving this a bit vague for the moment. I might barely have enough time to get this submitted, or I might have more than enough time. In that case, I hope to develop the preview/output to HTML+CSS/Excel further to incorporate a user interface/preview option.
+
```

Comment:
I think your plan has a nice balance of knowing what milestones you need to complete, while also having some flexibility if you need it. The work you do this first week in figuring out which host language and libraries you should use is probably going to end up being really important.